rclcpp_components/CMakeLists.txt

```diff
@@ -22,6 +22,7 @@ find_package(rcpputils REQUIRED)
 
 # Add an interface library that can be dependend upon by libraries who register components
 add_library(component INTERFACE)
+add_library(rclcpp::component ALIAS component)
```

Comment:
Should `rclcpp` be `${PROJECT_NAME} `?

Reply:
I had pondered this, and believe it is up to the project maintainers to define a convention. See this excerpt from Craig Scott's book, Professional CMake.

```
Target names need not be related to the project name. It is common to see tutorials and examples
use a variable for the project name and reuse that variable for the name of an executable target.

This only works for the most basic of projects and encourages a number of bad habits. Consider the
project name and executable name as being separate, even if initially they start out the same. Set
the project name directly rather than via a variable, choose a target name according to what the
target does rather than the project it is part of and assume the project will eventually need to define
more than one target. This reinforces better habits which will be important when working on more
complex multi-target projects.
```

Although it is mentioning executable, this practice is very common in ROS2 for libraries being named the same as the project.  I have not seen a convention for ALIAS libraries yet, other than 90% of ROS libraries name a single target the same as the project, which does not seem to be recommended. It also would break ABI to change existing library target names, however, it seems a  good time to pick convention for ALIAS libs.  As far as a naming convention, it ideally can handle the following scenarios. 

- A ROS2 repo with a single ROS package that has single target
- A ROS2 repo with a single ROS package that has multiple targets (executables, libraries, etc)
- A ROS2 repo with multiple ROS packages that have multiple target(s)

Let me know your ideas, and I can update this PR to match, as well as perhaps clarify the convention in tutorials/docs if accepted.

Reply:
This should be `rclcpp_components::component`.
That's also the exported target name